### DISCUSSION TITLE
Equispaced SO3 Grid

### DISCUSSION THREAD

**Discussion by grld** [mtex-toolbox/mtex#1864] [Ask Anything] [answered]
Dear community,

I posted this question two weeks ago under #1845. My issue is that I cannot fully reconstruct how the grid is created in MTEX. If anyone here has more experience with the Equispaced SO3 Grid method or might know the person who implemented that in MTEX, I would appreciate any help! 

Kind regards.

**Comment by ralfHielscher** [comment]
I implemented it. What do you want to know?

Ralf.

**Reply by grld** [reply]
Dear Ralf,

thanks for getting back to me. I am reading the habilitation thesis of Kurt Helming to understand more about the nearly equal distant grid in the orientation space. The first thing that he does is create a so-called equal-area-raster on the sphere in which the grid points are the center points of almost equal-sized surface elements. Intuition of this grid is given in Figure 7 of his work:
![Figure 7](https://github.com/mtex-toolbox/mtex/assets/39367797/72569428-589b-4728-a35f-2ec9c0e8f92e)
When I inspect your code, you follow a similar approach and first create the spherical grid with equispacedS2Grid(). You first determine the number of concentrical circles `ntheta`  and then for each circle of polar angle`th` (=k in Fig. 7), the number of points is determined with:
`steps = max(round(sin(th) * bounds.drho / bounds.dtheta * ntheta),1);`
Then the azimuthal angles $`\rho`$ are set on this circle by:
`bounds.VR{3} + (0:steps-1 )* bounds.drho /steps + mod(j,2) * bounds.drho/steps/2`,
where `bounds.VR{3}` is the minimum bound of $`\rho`$ and  `drho` its interval in the fundamental zone.
**Q1:** Does this result in the same distribution of points on a sphere as the approach of Helming? I do not understand the equation where the number of steps (=azimuthal angles $`\rho`$ on a circle) is determined and why for every second circle, a shift of `bounds.drho/steps/2` is applied.

After this step, I assume we have an equal-area raster on the sphere, similar to subfigure c) in Fig. 7. (not like a))
In the following step, the third angle is determined to complete the transformation from S2 to SO3 grid. In the code, this section is referred to as "_calculate gamma shift_". I can not follow the approach there.
**Q2:** How is the third Euler angle (gamma) determined here?

In order to derive the grid for SO3 instead of S2, Helming further argues in his work "_[...]An analysis of the formulas for the orientation distance_ $`\omega`$ _shows that an approximately equidistant grid in the orientation space cannot be produced by discretizing the usual orientation parameters_ $`\phi_{1}, \Phi, \phi_{2}`$." Due to that fact, he proposes to decompose the orientation g  into two successive rotations that are formulated in axis-angle descriptions. This procedure is described in more detail in a second publication by Helming [here](https://doi.org/10.1155/TSM.28.219). The first rotation brings the specimen coordinate system into an intermediate state by rotating by $`\sigma = \frac{\phi_{1}+\phi_{2}}{2}`$ around the specimen Z axis $`\text{Z}_{A}`$. The second rotation transforms from the intermediate state to the final crystal coordinate system. Helming describes the creation of the grid in SO3 then as a two-step procedure: 
1) Each direction of the S2Grid is identified as a different direction $`\text{Z}_{A}`$-direction
2) For each $`\text{Z}_{A}`$-direction, a number of $`\sigma`$ sections are sampled

**Q3:** Is this approach similar to your way of sampling the gamma angle for each spherical coordinate of the S2 grid created in step 1?

The habilitation thesis (in German) can be found [here](https://www.researchgate.net/profile/Kurt-Helming/publication/35447875_Texturapproximation_durch_Modellkomponenten/links/5899d73c4585158bf6f86af8/Texturapproximation-durch-Modellkomponenten.pdf). The pages I am referring to are 23-24. 

Thank you very much for your help in advance!

**Reply by ralfHielscher** [reply, marked as answer]
The approach by MTEX is indeed very similar to the approach of Helming. The sphere representing two of the three Euler angles is covered by an almost equispaced grid and the third Euler angle sampled equidistantly as well.

> Q1: Does this result in the same distribution of points on a sphere as the approach of Helming? I do not understand the equation where the number of steps (=azimuthal angles on a circle) is determined and why for every second circle, a shift of bounds.drho/steps/2 is applied.

The number of points on the small circles is determined by the length of the small circle divided by the spacing in the theta angle. The shift has been proven to make the points a little bit better equispaced. 

> After this step, I assume we have an equal-area raster on the sphere, similar to subfigure c) in Fig. 7. (not like a))
> In the following step, the third angle is determined to complete the transformation from S2 to SO3 grid. In the code, this section is referred to as "calculate gamma shift". I can not follow the approach there.
> Q2: How is the third Euler angle (gamma) determined here?
The third Euler angle is sampled simply equidistantly. The shift has no influence on the grid. It is only to ensure the orientations are within the fundamental region. The orientation grid within MTEX is fully characterized by being the tensor product of an equidistribution on the sphere and an equidistribution with respect to the third Euler angle.

The grids provided by MTEX are approximately equispaced. In fact there is no perfect equispaced grid on SO(3). For that reason many people came up with more or less good approximations. I think for most use cases the grid provided by MTEX is reasonably good. It works for all combination of symmetries and allows for quickly finding neighboring points. 

I hope this helps,

Ralf.
